File: data/code_pairs/cpp/1028.cpp
// PermanentArrays.hpp
#include <vector>
#include <string>

class PermanentArrays {
public:
    std::vector<int> labels;
    std::vector<double> xparam;
    std::vector<std::string> simbol;
};

// test_permanent_arrays.cpp
#include "PermanentArrays.hpp"
#include <iostream>

int main() {
    PermanentArrays pa;

    pa.labels = {1, 2, 3, 4, 5};
    pa.xparam = {1.0, 2.0, 3.0, 4.0, 5.0};
    pa.simbol = {"H   ", "He  "}; // Space padded to match Fortran's fixed length

    std::cout << "Labels: ";
    for (const auto& label : pa.labels) {
        std::cout << label << " ";
    }
    std::cout << std::endl;

    std::cout << "XParam: ";
    for (const auto& x : pa.xparam) {
        std::cout << x << " ";
    }
    std::cout << std::endl;

    std::cout << "Simbol: ";
    for (const auto& s : pa.simbol) {
        std::cout << s << " ";
    }
    std::cout << std::endl;

    return 0;
}

File: data/code_pairs/fortran/1028.f90
! permanent_arrays.f90
module permanent_arrays
  implicit none
  integer, parameter :: double = selected_real_kind(15, 307)
  integer, dimension (:), allocatable :: labels
  real(double), dimension (:), allocatable :: xparam
  character(len=4), dimension (:), allocatable :: simbol
end module permanent_arrays

program test_permanent_arrays
  use permanent_arrays
  implicit none

  allocate(labels(5))
  allocate(xparam(5))
  allocate(simbol(2))

  labels = (/1, 2, 3, 4, 5/)
  xparam = (/1.0_double, 2.0_double, 3.0_double, 4.0_double, 5.0_double/)
  simbol(1) = "H   "
  simbol(2) = "He  "

  print *, "Labels: ", labels
  print *, "XParam: ", xparam
  print *, "Simbol: ", simbol

  deallocate(labels)
  deallocate(xparam)
  deallocate(simbol)
end program test_permanent_arrays